packages/guideline-blocks-settings/src/hooks/useAttachments.tsx

```diff
@@ -6,23 +6,18 @@ import { type ReactNode, createContext, useContext } from 'react';
 import { type BlockProps } from '../index';
 
 export const useAttachments = (appBridge: AppBridgeBlock, assetId: string) => {
-    const { blockAssets, updateAssetIdsFromKey } = useBlockAssets(appBridge);
+    const { blockAssets, addAssetIdsToKey, deleteAssetIdsFromKey, updateAssetIdsFromKey } = useBlockAssets(appBridge);
     const attachments = blockAssets?.[assetId] || [];
 
     const onAttachmentsAdd = async (newAssets: Asset[]) => {
-        const newAssetIds = attachments.map((attachment) => attachment.id);
-        for (const asset of newAssets) {
-            newAssetIds.push(asset.id);
-        }
-        await updateAssetIdsFromKey(assetId, newAssetIds);
+        await addAssetIdsToKey(
+            assetId,
+            newAssets.map((asset) => asset.id),
+        );
     };
 
     const onAttachmentDelete = async (assetToDelete: Asset) => {
-        const newAssetIds = attachments
-            .filter((attachment) => attachment.id !== assetToDelete.id)
-            .map((attachment) => attachment.id);
-
-        await updateAssetIdsFromKey(assetId, newAssetIds);
+        await deleteAssetIdsFromKey(assetId, [assetToDelete.id]);
```

Comment:
Looking at the function names should we rename the argument `assetId` in the useAttachments hook? it seems odd to call deleteAssetIdsFromKey with the first argument being `assetId`, which is actually the `key` in this case

Reply:
changed to `attachmentKey`